File: go-backend/internal/http/middleware/cors.go
package middleware

import (
	"task-system-go/internal/config"

	"github.com/gin-gonic/gin"
)

func CORS(cfg config.Config) gin.HandlerFunc {
	return func(c *gin.Context) {
		c.Writer.Header().Set("Access-Control-Allow-Origin", cfg.AllowOrigin)
		c.Writer.Header().Set("Access-Control-Allow-Methods", "GET,POST,PUT,PATCH,DELETE,OPTIONS")
		c.Writer.Header().Set("Access-Control-Allow-Headers", "Authorization,Content-Type")
		if c.Request.Method == "OPTIONS" {
			c.AbortWithStatus(204)
			return
		}
		c.Next()
	}
}

File: backend-tp8/app/middleware/Cors.php
<?php

declare(strict_types=1);

namespace app\middleware;

use think\Request;
use think\Response;

class Cors
{
    public function handle(Request $request, \Closure $next): Response
    {
        if ($request->isOptions()) {
            return response('', 204)->header($this->headers());
        }
        
        return $next($request)->header($this->headers());
    }

    private function headers(): array
    {
        return [
            'Access-Control-Allow-Origin' => '*',
            'Access-Control-Allow-Methods' => 'GET,POST,PUT,PATCH,DELETE,OPTIONS',
            'Access-Control-Allow-Headers' => 'Authorization,Content-Type',
        ];
    }
}
